### PR TITLE
docs: fix measure-comparison demo's corner cell's style and last col's line style

### PR DESCRIPTION
**Pull request by yangzhanmei**
### 👀 PR includes

🔧 Chore

- [ ] Test case
- [x] Docs / demos update
- [ ] CI / workflow
- [ ] Release version
- [ ] Other ()

### 📝 Description
docs: fix measure-comparison demo's corner cell's style and last col's line style
### 🖼️ Screenshot

| Before                         | After                         |
| ------------------------------ | ------------------------------ |
| ❌                             | ✅                              |
|      image                       |          image                    |
| image                            | image                             |

### 🔍 Self Check before Merge

- [ ] Add or update relevant Docs.
- [ ] Add or update relevant Demos.
- [ ] Add or update relevant TypeScript definitions.

### CODE REVIEW

**Review by YardWill — DISMISSED** [review]

**Review by qubaomingg — DISMISSED** [review]
LG

File: s2-site/examples/case/measure-comparison/demo/basic.tsx [hunk 1]
```diff
@@ -280,7 +281,11 @@ class CustomDataCell extends DataCell {
         },
       });
     }
-    if ((colIndex + 1) % valueLength === 0) {
+    const tagLength = [...(spreadsheet.dataSet.colPivotMeta || [])].length;
+    if (
+      (colIndex + 1) % valueLength === 0 &&
+      colIndex + 1 !== tagLength * valueLength  // 最后一列不加line
+    ) {
```
Comment (stone-lyl) [inline s2-site/examples/case/measure-comparison/demo/basic.tsx:288]: if 括号中的内容代表什么含义呢？能否添加注释 或者将其提取为一个语义化的方法

File: s2-site/examples/case/measure-comparison/demo/basic.tsx [hunk 2]
```diff
@@ -389,6 +394,17 @@ class CustomCornelCell extends CornerCell {
     }
     super.drawCellText();
   }
+  drawBackgroundShape() {
+    const { backgroundColorOpacity, backgroundColor } = this.getStyle().cell;
+    const attrs = {
+      ...this.getCellArea(),
+      fill: this.meta.cornerType === 'col' ? '#FFFFFF' : backgroundColor,
```
Comment (stone-lyl) [inline s2-site/examples/case/measure-comparison/demo/basic.tsx:401]:
```suggestion
      fill: this.meta.cornerType === 'col' ? '#FFF' : backgroundColor,
```

File: s2-site/examples/case/measure-comparison/demo/basic.tsx [hunk 3]
```diff
@@ -280,7 +281,11 @@ class CustomDataCell extends DataCell {
         },
       });
     }
-    if ((colIndex + 1) % valueLength === 0) {
+    const tagLength = [...(spreadsheet.dataSet.colPivotMeta || [])].length;
```
Comment (wjgogogo) [inline s2-site/examples/case/measure-comparison/demo/basic.tsx:284]: 这里只为了获取length，有必要解构复制一下吗？

Comment (yangzhanmei) [reply s2-site/examples/case/measure-comparison/demo/basic.tsx:284]: meta是个map

**Review by xingwanying — APPROVED** [review]
LG

**Pull request merged**